src/mca/bfrops/base/base.h

```diff
@@ -158,47 +160,306 @@ PMIX_EXPORT extern pmix_bfrops_globals_t pmix_bfrops_globals;
 #error Unsupported pid_t size!
 #endif
 
+/* Flexible packing */
+#define PMIX_BFROPS_FLEX_BASE7_MAX_BUF_SIZE (SIZEOF_SIZE_T+1)
+#define PMIX_BFROPS_FLEX_BASE7_MASK ((1<<7) - 1)
+#define PMIX_BFROPS_FLEX_BASE7_SHIFT 7
+#define PMIX_BFROPS_FLEX_BASE7_CONT_FLAG (1<<7)
+
+/**
+ * Packing conversion of a signed integer value to a flexible representation.
+ * The main idea is to split a signed negative value onto an absolute value
+ * and a sing bit stored in the special location.
```

Comment:
```suggestion
 * and a sign bit stored in the special location.
```

Reply:
I didn't know you could do that - neat!!

Reply:
A trick I learned at the MPI Forum. I'll show you sometime :)